File: MIniProjekPRG7 - BackEnd/src/main/java/com/example/miniprojekprg7/service/SparepartService.java
package com.example.miniprojekprg7.service;

import com.example.miniprojekprg7.response.DtoResponse;
import com.example.miniprojekprg7.vo.SparepartVoForm;

public interface SparepartService {
    DtoResponse getAllSpareparts();
    DtoResponse getSparepartActive();
    DtoResponse getSparepartById(int id);

    DtoResponse saveSparepart(SparepartVoForm sparepart);

    DtoResponse updateSparepart(SparepartVoForm sparepart, int id);

    DtoResponse deleteSparepart(int id);
}

File: MiniProjekPRG7 - FrontEnd/miniproject-sparepart/src/service/SparepartService.js
import axios from "axios";

const REST_API_BASE_URL = "http://localhost:8080/spareparts";

export const listSpareparts = () => axios.get(REST_API_BASE_URL + "/getAllSpareparts");
export const saveSparepart = (sparepat) => axios.post(REST_API_BASE_URL + "/saveSparepart", sparepat);
export const getSparepartById = (id) => axios.get(REST_API_BASE_URL + `/getSparepartById/${id}`);
export const updateSparepart = (sparepat, id) => axios.post(REST_API_BASE_URL + `/updateSparepart/${id}`, sparepat);
export const deleteSparepart = (id) => axios.post(REST_API_BASE_URL + `/deleteSparepart/${id}`);
